domino/domino.py

```diff
@@ -525,15 +525,41 @@ def collaborators_get(self):
         url = self._routes.collaborators_get()
         return self._get(url)
 
-    def collaborators_add(self, usernameOrEmail, message=""):
+    def get_user_id(self, username_or_email): 
+        url = self._routes.users_get()
+        response = self.request_manager.get(url)
+        users = response.json()
+        user_from_user_name = [x for x in users if ('userName' in x) and (x['userName'] == username_or_email)]
+        user_from_email = [x for x in users if ('email' in x) and (x['email'] == username_or_email)]
+        user_list = user_from_user_name + user_from_email
```

Comment:
Apologies for not explaining my thoughts in the longer comment earlier @ddl-abayly -- I was trying to finish up two other tasks, and wanted to post something quickly in case you were waiting to merge something by the end of the day. With a bit more time, I can expand on it a bit. (Also, with hindsight, I fixed a glaring bug in my earlier suggestion.)

I think the three lines here are doing more work than you necessary. 

* Each list comprehension iterates through the `users` list independently, while the `for` loop iterates through it only once.
* You don't need to check "if key in user" if you use `user.get(key)`. (This was my bug. I didn't realize that some users don't have email accounts -- like the default Jira user. Accessing the key like this won't throw a `KeyError`.)
* The original approach creates two (albeit short, or even empty) lists, and then a third by concatenating those, just to get at the first element -- which all seemed busy, and a bit hard to read.

Of course, my suggested edit assumes that users have unique usernames and email address (which seems like a reasonable assumption).